src/BizHawk.Client.Common/movie/import/HexenLmpImport.cs

```diff
@@ -0,0 +1,132 @@
+
+namespace BizHawk.Client.Common
+{
+	// LMP file format: https://doomwiki.org/wiki/Demo#Technical_information
+	// In better detail, from archive.org: http://web.archive.org/web/20070630072856/http://demospecs.planetquake.gamespy.com/lmp/lmp.html
+	[ImporterFor("Hexen", ".hexenlmp")]
+	internal class HexenLmpImport : MovieImporter
+	{ 
+		protected override void RunImport()
+		{
+			Result.Movie.HeaderEntries[HeaderKeys.Core] = CoreNames.DSDA;
+			var platform = VSystemID.Raw.Doom;
+			var settings = new DSDA.DoomSettings();
+			var syncSettings = new DSDA.DoomSyncSettings();
+
+			Result.Movie.SystemID = platform;
+
+			// Getting input file stream
+			var input = SourceFile.OpenRead().ReadAllBytes();
+			Stream sr = new MemoryStream(input);
+
+			// Parsing header
+			byte skillLevel = (byte)sr.ReadByte();
+			byte episode = (byte) sr.ReadByte();
+			byte map = (byte) sr.ReadByte();
+			byte player1Present = (byte) sr.ReadByte();
+			byte player1Class = (byte) sr.ReadByte();
+			byte player2Present = (byte) sr.ReadByte();
+			byte player2Class = (byte) sr.ReadByte();
+			byte player3Present = (byte) sr.ReadByte();
+			byte player3Class = (byte) sr.ReadByte();
+			byte player4Present = (byte) sr.ReadByte();
+			byte player4Class = (byte) sr.ReadByte();
+
+			// Players [5-8] ignored
+			byte player5Present = (byte) sr.ReadByte();
+			byte player5Class = (byte) sr.ReadByte();
+			byte player6Present = (byte) sr.ReadByte();
+			byte player6Class = (byte) sr.ReadByte();
+			byte player7Present = (byte) sr.ReadByte();
+			byte player7Class = (byte) sr.ReadByte();
+			byte playerPresent = (byte) sr.ReadByte();
+			byte player8Class = (byte) sr.ReadByte();
+
+			// Setting values
+			syncSettings.InputFormat = DoomControllerTypes.Hexen;
+			syncSettings.SkillLevel = (DSDA.SkillLevelEnum) (skillLevel+1);
+			syncSettings.InitialEpisode = episode;
+			syncSettings.InitialMap = map;
+			syncSettings.MultiplayerMode = DSDA.MultiplayerModeEnum.M0;
+			syncSettings.MonstersRespawn = false;
+			syncSettings.FastMonsters = false;
+			syncSettings.NoMonsters = false;
+			settings.DisplayPlayer = 0;
+			syncSettings.Player1Present = player1Present > 0;
+			syncSettings.Player1Class = (DSDA.HexenClassEnum) player1Class;
+			syncSettings.Player2Present = player2Present > 0;
+			syncSettings.Player2Class = (DSDA.HexenClassEnum) player2Class;
+			syncSettings.Player3Present = player3Present > 0;
+			syncSettings.Player3Class = (DSDA.HexenClassEnum) player3Class;
+			syncSettings.Player4Present = player4Present > 0;
+			syncSettings.Player4Class = (DSDA.HexenClassEnum) player4Class;
+			syncSettings.CompatibilityMode = DSDA.CompatibilityLevelEnum.C0;
+
+			var hexenController = new HexenController(0);
+			var controller = new SimpleController(hexenController.Definition);
+			controller.Definition.BuildMnemonicsCache(Result.Movie.SystemID);
+
+			bool isFinished = false;
+			while (!isFinished)
+			{
+				if (player1Present > 0) parsePlayer(controller, sr, 0);

```

Comment:
Why is this indexed from 0?

Reply:
I don't have a good answer for this. For some reason, I need to pass playerId 0 for P1 inputs. If I use [1-4] indexing, it'll be shifted by one. I haven't done anything in DSDA to warrant this, it just works like that, so I assume it's internal to BK?

Reply:
https://github.com/TASEmulators/BizHawk/blob/699c61a131798cfd3275272bf2ab5fc6890490ee/src/BizHawk.Emulation.Cores/Computers/Doom/DSDA.ControllerDeck.cs#L14
https://github.com/TASEmulators/BizHawk/blob/699c61a131798cfd3275272bf2ab5fc6890490ee/src/BizHawk.Client.Common/movie/import/HexenLmpImport.cs#L70
?